llvm/lib/Transforms/IPO/ModuleInliner.cpp

```diff
@@ -242,13 +260,22 @@ PreservedAnalyses ModuleInlinerPass::run(Module &M,
           // iteration because the next iteration may not happen and we may
```

Comment:
Without your patch, if `ICB->getCalledFunction() != nullptr`, then you would do:

```
        if (NewCallee)
          if (!NewCallee->isDeclaration())
            Calls->push({ICB, NewHistoryID});
```

With your patch but without the contexual profile, that behavior seems to be lost.  Is that intended?

Reply:
removed this part of the change - didn't read `tryPromote` in detail and incorrectly assumed it was profile-driven ICP. Don't really need ICP-ing now either, the upfront one is sufficient (since this is mostly for test purposes at this point)